aws/resource_aws_ecs_service.go

```diff
@@ -495,24 +503,17 @@ func resourceAwsEcsServiceUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
-	if d.HasChange("network_configration") {
-		input.NetworkConfiguration = expandEcsNetworkConfigration(d.Get("network_configuration").([]interface{}))
+	if d.HasChange("network_configuration") {
+		input.NetworkConfiguration = expandEcsNetworkConfiguration(d.Get("network_configuration").([]interface{}))
 	}
 
-	// Retry due to IAM & ECS eventual consistency
+	// Retry due to IAM eventual consistency
 	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
 		out, err := conn.UpdateService(&input)
 		if err != nil {
-			awsErr, ok := err.(awserr.Error)
-			if ok && awsErr.Code() == "InvalidParameterException" {
-				log.Printf("[DEBUG] Trying to update ECS service again: %#v", err)
-				return resource.RetryableError(err)
-			}
-			if ok && awsErr.Code() == "ServiceNotFoundException" {
```

Comment:
Here we're losing the retry on `ServiceNotFoundException`, is that intentional?

Reply:
@catsby Interestingly enough, I had just assumed it was some sort of typo that it was included and that retrying on a missing service would be undesirable during an update, but turns out we are currently always calling `resourceAwsEcsServiceUpdate` at the end of `resourceAwsEcsServiceCreate`. So that explains why it was in there.

Looking through the parameters set during `resourceAwsEcsServiceUpdate` to `ecs.UpdateServiceInput`, I don't see anything not already handled by `resourceAwsEcsServiceCreate` so I think we should instead return `resourceAwsEcsServiceRead` at the end of `resourceAwsEcsServiceCreate` and keep the `ecs.ErrCodeServiceNotFoundException` retry logic out of `resourceAwsEcsServiceUpdate`. 

Does that sound reasonable?

Reply:
@bflad that sounds like a good idea - as long as all ECS tests still pass 🤷‍♂️